File: Java/src/main/java/Solution201_300/Solution263.java
package Solution201_300;

/**
 * @author windows
 */
public class Solution263 {
    public boolean isUgly(int n) {
        while(n != 1){
            if (n % 2 == 0){
                n /= 2;
            }else if (n % 3 == 0){
                n /= 3;
            }else if (n % 5 == 0){
                n /= 5;
            }else {
                return false;
            }
        }
        return true;
    }


    public static void main(String[] args) {
        Solution263 solution263 = new Solution263();
    }
}

File: C++/src/Solution201_300/Solution263.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution263.h"
#include <functional>
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

bool isUgly(int n) {
  if (n <= 0) return false;
  while (n != 1){
    if (n % 2 == 0){
      n /= 2;
    }else if (n % 3 == 0){
      n /= 3;
    }else if (n % 5 == 0){
      n /= 5;
    }else{
      return false;
    }
  }
  return true;
}